nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,799 @@
+                    inverse_prob = -decision
+            score = scorer(estimator, X_test, y_test)
+            if np.all(estimator.coef_ == 0):
+                score = 0
+        else:  # regression
+            y_prob = estimator.predict(X_test)
+            score = scorer(estimator, X_test, y_test)
+
+        if (best_score is None) or (score >= best_score):
+            best_score = score
+            best_coef = estimator.coef_
+            best_intercept = estimator.intercept_
+            best_y = {}
+            best_y['y_prob'] = y_prob
+            best_y['y_true_indices'] = y_true_indices
+            best_param = param
+            if is_classif:
+                best_y['inverse'] = inverse_prob
+
+    if selector is not None:
+        best_coef = selector.inverse_transform(best_coef)
+
+    if LooseVersion(sklearn.__version__) <= LooseVersion('0.17'):
+        # scikit-learn > 0.17
+        if isinstance(estimator, SVR):
+            best_coef = -best_coef
+
+    return (class_index, best_coef, best_intercept, best_y, best_param,
+            best_score)
+
+
+class BaseDecoder(LinearModel, RegressorMixin, CacheMixin):
+    """A wrapper for popular classification/regression strategies for
+    neuroimgaging.
+
+    The `BaseDecoder` object supports classification and regression methods.
+    It implements a model selection scheme that averages the best models
+    within a cross validation loop. The resulting average model is the
+    one used as a classifier or a regressor. The `Decoder` object also
+    leverages the `NiftiMaskers` to provide a direct interface with the
+    nifti files on disk.
+
+    Parameters
+    -----------
+    estimator : str, optional
+        The estimator to choose among: 'svc', 'svc_l1', 'logistic',
+        'logistic_l1', 'ridge_classifier', 'ridge_regression',
+        and 'svr'. Defaults to 'svc'.
+
+    mask: filename, NiImage, NiftiMasker, or MultiNiftiMasker, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a masker with default
+        parameters.
+
+    cv : cross-validation generator, optional (default 10)
+        A cross-validation generator. If None, a 3-fold cross
+        validation is used for regression or 3-fold stratified
+        cross-validation for classification.
+
+    param_grid : dict of string to sequence, or sequence of such
+        The parameter grid to explore, as a dictionary mapping estimator
+        parameters to sequences of allowed values.
+
+        An empty dict signifies default parameters.
+
+        A sequence of dicts signifies a sequence of grids to search, and is
+        useful to avoid exploring parameter combinations that make no sense
+        or have no effect. See scikit-learn documentation for more information.
+
+    screening_percentile: int, float, optional, in the closed interval [0, 100]
```

Comment:
I think some details are missing here. From seeing the same parameter in nilearn._utils.param_validation.py in _adjust_screening_percentile. AFAIK adding more details at object level would help for users.